crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -308,12 +306,24 @@ impl C8YHttpProxyActor {
             download_info.auth = Some(Auth::new_bearer(token.as_str()));
         }
 
-        // Download a file to tmp dir
-        let file_name = request.file_path.file_name().unwrap().to_str().unwrap();
-        let downloader: Downloader = Downloader::new_sm(file_name, &None, self.tmp_dir.clone());
+        // Download file to the target directory with a temp name
+        let file_name = nanoid!();
+        let parent_dir = request.file_path.parent().ok_or_else(|| {
+            C8YRestError::CustomError(format!(
+                "Parent directory of {:?} not found",
+                request.file_path
+            ))
+        })?;
+        debug!(target: self.name(), "Downloading from: {:?}", download_info.url());
+        let downloader: Downloader = Downloader::new_sm(&file_name, &None, parent_dir);
         downloader.download(&download_info).await?;
 
         // Move the downloaded file to the final destination
+        debug!(
+            "Moving downloaded file from {:?} to {:?}",
+            downloader.filename(),
+            request.file_path
+        );
```

Comment:
> However, after downloading a file, 99% of usecases requires to move a file from location A to B.

I would even say 100% of the time as otherwise one has no clue is a file is the result of a complete download or some garbage left by an aborted download.

Reply:
Deferred to a follow-up PR to avoid touching other components using the downloader (tedge-agent and firmware-plugin).